src/components/roundHistory/RoundHistoryHooks.ts

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useState } from "react";
+import { GameState } from "../../const/gameState";
+import { winner } from "../../const/gameWinner";
+import gameContextHelpers from "../../context/GameContextHelper";
+import useGameContext from "../../hooks/useGameContext";
+import { EndedRoundStatus } from "../../types/global";
+import pointsHelpers from "../points/PointsHelper";
+
+export const useRoundHistory = () => {
+  const [previousRoundsDetails, setPreviousRoundsDetails] = useState<EndedRoundStatus[]>([]);
+  const [roundWinners, setRoundWinners] = useState<string[]>([]);
+
+  const { playerCards, croupierCards, setMessage, currentRoundStatus } = useGameContext();
+
+  useEffect(() => {
+    if (currentRoundStatus === GameState.FINISH_ROUND) {
+      const gameWinner = gameContextHelpers.findWhoWonRound(
+        pointsHelpers.getPlayerPoints(playerCards),
+        pointsHelpers.getCroupierPoints(croupierCards, true)
+      );
+
+      switch (gameWinner) {
```

Comment:
mógłbyś użyć object literal zamiast switcha, zazwyczaj dobrze to wpływa na czytelność kodu (https://ultimatecourses.com/blog/deprecating-the-switch-statement-for-object-literals)
```suggestion
      const message = gameResultMessage[winner.PLAYER];
      setMessage(message);
```